lib/nats.js

```diff
@@ -843,6 +865,9 @@ Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
  */
 
 Client.prototype.subscribe = function(subject, opts, callback) {
+  if (this.closed) {
+    this.emit('error', SUB_CONN_CLOSED_ERR);
```

Comment:
Regarding the read loop, I could process each message individually, but even then event loop starvation is possible and the performance for applications that behave correctly would suffer. I think the yieldTime is a good alternative.